compiler/stable_mir/src/mir/body.rs

```diff
@@ -602,9 +663,9 @@ pub struct Constant {
 }
 
 #[derive(Clone, Debug, Eq, PartialEq)]
-pub struct SwitchTarget {
-    pub value: u128,
-    pub target: usize,
+pub struct SwitchTargets {
+    pub value: Vec<u128>,
+    pub targets: Vec<usize>,
```

Comment:
Why did you change this? I usually prefer one vector of pairs than a pair of vectors, since the first makes it explicit which values should always go together.

Reply:
Because we could have multiple(2) targets for one value, see here https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/mir/struct.SwitchTargets.html, they are always one pair, and I found this to be easier to reason about.

Reply:
Gotcha, I did notice the following comment that I think we should consider here:

https://github.com/rust-lang/rust/blob/3166210d68fe43161ef64d682dd604cc60e7607f/compiler/rustc_middle/src/mir/syntax.rs#L836-L844

Reply:
🤔 would this really matter for our case, and they haven't found a case where it's better than the one they have 😄 what do you recommend ?

Reply:
I thnk in the rust compiler case, it makes sense to avoid a big refactoring, but for our case, I think the structure suggested in the comment is cleaner and less error prone. I would prefer that unless we have a stronger reason not to.

Reply:
Hmm found quite annoying bug, currently we never print `otherwise`  because we have wrong number of successors, e.g for this `[0,1,"otherwise"]` and successors count would be `3` in `stable_mir` it's `2` this might be due to fact we don't have some of the `TerminatorKind` variants in `stable_mir` 😕

Reply:
I think the problem is with the conversion from MIR to StableMIR. We're not translating the otherwise target.

If you do prefer keeping the structure as is, can you please make the members private and provide methods to read and write them please?

Reply:
As discussed offline, we can work on a better structure for SwitchTargets in a follow up PR. This change is however better than our current approach which is missing `otherwise`, so let's move ahead with this for now.